### ISSUE TITLE
Audit existing code for ad-hoc environment discovery.

### ISSUE THREAD

**Issue by ndouglas** [department-of-veterans-affairs/va.gov-cms#13846]
## Description

We currently have an unknown amount of custom code that performs ad-hoc environment discovery in a way that is not easily testable and is tightly coupled to the actual environment.

After #13845 is completed, we should audit the codebase for code that performs this sort of work and open issues to transition it to a better design.

## Acceptance Criteria

- [x] Custom code in the following areas has been audited for ad-hoc environment discovery:
  - [x] custom modules
  - [x] custom themes
  - [x] scripts
- [x] Determinations have been made as to which of these discovery operations are problematic and worth rewriting, and are documented in this thread.
- [x] Followup issues have been opened where appropriate to rewrite the code.
- [x] A followup issue has been opened to address any feature deficits in the environment discovery service that would assist in completing the followup issues.

**Comment by ndouglas** [comment]
`docroot/modules/custom/va_gov_backend.module`:

```php
/**
 * Implements hook_page_bottom().
 */
function va_gov_backend_page_bottom(array &$page_bottom) {
  // Add GTM noscript iframe to body.
  $env_name = \Drupal::config('environment_indicator.indicator')->get('name');
  $env_num = $env_name === 'Production' ? '2' : '5';
  $gtm_noscript = Markup::create('<noscript><iframe src="https://www.googletagmanager.com/ns.html?id=GTM-WQ3DLLB&gtm_auth=hHvHnY1eIx1IOSJMAnOBLA&gtm_preview=env-' . $env_num . '&gtm_cookies_win=x" height="0" width="0" style="display:none;visibility:hidden"></iframe></noscript>');
  $page_bottom['va_gov_backend'] = ['#markup' => $gtm_noscript];
}
```

```php

/**
 * Implements hook_rebuild().
 */
function va_gov_backend_rebuild() {
  // @todo refactor this to use \Drupal\va_gov_consumers\Git\Git.
  // Update the drupal state to display git info in admin toolbar.
  $environment = \Drupal::config('environment_indicator.indicator')->get('name');
  $branch = mb_strimwidth(exec('git rev-parse --abbrev-ref HEAD'), 0, 20, '...') ?? 'git unknown';
  $commit_hash = trim(exec('git rev-parse --short HEAD'));
  $git_tag = exec("git tag --points-at $commit_hash");
  // Remove the 'cms/' from the tag that looks like 'cms/v0.0.190'.
  $git_tag = str_replace('cms/v', '', $git_tag);

  $brd = [
    'Development',
    'Staging',
    'Production',
  ];

  if (($environment === 'Production') && !empty($git_tag)) {
    $indicator = "v{$git_tag}";
  }
  elseif (in_array($environment, $brd)) {
    // We are in a BRD environment but with no tag.
    $indicator = "v{$commit_hash}";
  }
  else {
    // We are in not in a BRD environment, so display branch and SHA.
    $indicator = "Branch: {$branch}  v{$commit_hash}";
  }

  \Drupal::state()->set("environment_indicator.current_release", $indicator);

  $environment_label = '';
  $env = getenv('CMS_ENVIRONMENT_TYPE') ?: 'ci';
  switch ($env) {
    case 'prod':
      $environment_label = 'Production Environment';
      break;

    case 'staging':
      $environment_label = 'Staging Environment';
      break;

    case 'dev':
      $environment_label = 'Dev Environment';
      break;

    case 'tugboat':
      $tugboat = getenv('TUGBOAT_PREVIEW_TYPE');
      if ($tugboat === 'pullrequest') {
        $environment_label = 'PR Environment';
      }
      elseif ($tugboat === 'branch') {
        $environment_label = 'Demo Environment';
      }
      break;

    case 'local':
      $environment_label = 'Local Environment';
      break;

  }

  \Drupal::state()->set("environment_indicator.environment_label", $environment_label);
  \Drupal::state()
    ->set('environment_indicator.env', $env);
}
```

```php
  ...
  $white_font = ['prod', 'local', 'dev'];
  if (in_array(\Drupal::state()
    ->get('environment_indicator.env'), $white_font)) {
    $items['content_release_icon'] = [
      '#attached' => [
        'library' => [
          'va_gov_backend/tippy_popover_theme',
          'va_gov_backend/toolbar',
        ],
      ],
      '#type' => 'toolbar_item',
      'tab' => [
        '#type' => 'html_tag',
        '#tag' => 'button',
        '#value' => '<span class="button-text">Last Content Release Time</span>',
        '#attributes' => [
          'class' => ['toolbar-icon', 'toolbar-icon-content-release'],
          'title' => 'Mouse over to see last content release time',
          'id' => 'content-release-status-icon',
        ],
      ],
      '#wrapper_attributes' => [
        'class' => ['toolbar-tab', 'content-release-toolbar-tab-prod'],
      ],
      '#weight' => 102,
    ];
  }
  ...
```

I think these are all problematic.

**Comment by ndouglas** [comment]
- `docroot/modules/custom/va_gov_backend/src/Service/Datadog.php`: Not a priority, realistically.
- `docroot/modules/custom/va_gov_build_trigger/src/Form/PreviewForm.php`: Ugh.
- `docroot/modules/custom/va_gov_user/src/EventSubscriber/UserImport.php`: Yeah, soon.

**Comment by ndouglas** [comment]
Opened followup issues:

- #14170 
- #14171 
- #14172 
- #14173
